src/psyclone/dynamo0p3.py

```diff
@@ -4399,111 +4347,79 @@ def max_depth_m1(self):
-        access depth, if one exists, and None if not
-        :param int literal_depth: The known fixed (literal) halo \
-        access depth
-        :param bool annexed_only: True if only the halo's annexed dofs \
-        are accessed and False otherwise
-        :param bool max_depth_m1: True if the field accesses all of \
-        the halo but does not require the outermost halo to be correct \
-        and False otherwise
+        :param bool max_depth: True if the field accesses all of the
+            halo and False otherwise
+        :param var_depth: PSyIR expression specifying the halo
+            access depth, if one exists, and None if not
+        :type var_depth: :py:class:`psyclone.psyir.nodes.Node`
+        :param bool annexed_only: True if only the halo's annexed dofs
+            are accessed and False otherwise
+        :param bool max_depth_m1: True if the field accesses all of
+            the halo but does not require the outermost halo to be correct
+            and False otherwise
 
         '''
         self._max_depth = max_depth
-        self._var_depth = var_depth
-        self._literal_depth = literal_depth
         self._annexed_only = annexed_only
         self._max_depth_m1 = max_depth_m1
+        self._var_depth = None
+
+        if not var_depth:
+            return
+
+        # Simplify any provided PSyIR expression.
+        sym_maths = SymbolicMaths.get()
+        # In order to be able to apply the SymbolicMaths.expand() method we
+        # have to create a fake Assignment and temporarily graft it into the
+        # tree.
+        fake_assign = Assignment.create(
+            Reference(DataSymbol("tmp", INTEGER_TYPE)), var_depth.detach())
+        sched = self._parent.ancestor(Schedule, include_self=True)
+        sched.addchild(fake_assign)
+
+        sym_maths.expand(fake_assign.rhs)
+        self._var_depth = fake_assign.rhs.detach()
+        fake_assign.detach()
 
     def __str__(self):
-        '''return the depth of a halo dependency
-        as a string'''
-        depth_str = ""
-        if self.max_depth:
-            max_depth = self._symbol_table.lookup_with_tag(
-                "max_halo_depth_mesh")
-            depth_str += max_depth.name
-        elif self.max_depth_m1:
-            max_depth = self._symbol_table.lookup_with_tag(
-                "max_halo_depth_mesh")
-            depth_str += f"{max_depth.name}-1"
-        else:
-            if self.var_depth:
-                depth_str += self.var_depth
-                if self.literal_depth:
-                    # Ignores depth == 0
-                    depth_str += f"+{self.literal_depth}"
-            elif self.literal_depth is not None:
-                # Returns depth if depth has any value, including 0
-                depth_str = str(self.literal_depth)
-        return depth_str
+        psyir = self.psyir_expression()
+        if psyir:
+            return FortranWriter()(psyir)
+        return "0"
```

Comment:
Just checking: is this still used outside a test? Because the lfric_loop.py you now use statements like:
`halo_index = FortranWriter()(self._upper_bound_halo_depth)`

Reply:
Oh yes, well spotted. I've deleted it now and updated the one test that used it.